transaction/_transaction.py

```diff
@@ -409,6 +416,67 @@ def _callAfterCommitHooks(self, status=True):
         self._after_commit = []
         self._before_commit = []
 
+    def getBeforeAbortHooks(self):
+        """ See ITransaction.
+        """
+        return iter(self._before_abort)
+
+    def addBeforeAbortHook(self, hook, args=(), kws=None):
+        """ See ITransaction.
+        """
+        if kws is None:
+            kws = {}
+        self._before_abort.append((hook, tuple(args), kws))
+
+    def _callBeforeAbortHooks(self):
+        # Call all hooks registered, allowing further registrations
+        # during processing.  Note that calls to addBeforeAbortHook() may
+        # add additional hooks while hooks are running, and iterating over a
+        # growing list is well-defined in Python.
+        for hook, args, kws in self._before_abort:
+            hook(*args, **kws)
+        self._before_abort = []
+
+    def getAfterAbortHooks(self):
+        """ See ITransaction.
+        """
+        return iter(self._after_abort)
+
+    def addAfterAbortHook(self, hook, args=(), kws=None):
+        """ See ITransaction.
+        """
+        if kws is None:
+            kws = {}
+        self._after_abort.append((hook, tuple(args), kws))
+
+    def _callAfterAbortHooks(self):
+        # Avoid to abort anything at the end if no hooks are registred.
+        if not self._after_abort:
+            return
+        # Call all hooks registered, allowing further registrations
+        # during processing.  Note that calls to addAterAbortHook() may
+        # add additional hooks while hooks are running, and iterating over a
+        # growing list is well-defined in Python.
+        for hook, args, kws in self._after_abort:
+            try:
+                hook(*args, **kws)
+            except:
+                # We need to catch the exceptions if we want all hooks
+                # to be called
+                self.log.error("Error in after abort hook exec in %s ",
+                               hook, exc_info=sys.exc_info())
+        # The transaction is already abortted. It must not have
```

Comment:
```suggestion
        # The transaction is already aborted. It must not have
```

Reply:
I have put the call hook logic into a centralized function (as you have recommended). As a consequence, the above comment disappeared.